hashintel/hash
# H-1465: Add graph visualization to `/entities` page

## 🌟 What is the purpose of this PR?




This PR adds a graph visualization to the `/entities` page.

## 🔗 Related links





- H-1465

## 🔍 What does this change?




- Moves the `EChart` component out of the `Chart` block and into the shared design-system package. Technically we should therefore republish the block, but as in follow up we are adding the `Graph` chart variant we can republish the block then.

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues



- Sometimes when switching between filters the graph hard resets. I'm not sure which of the properties is causing this, and why it only happens sometimes.

## 🐾 Next steps




## 🛡 What tests cover this?



None

## ❓ How to test this?



1. Go to the `/entities` page to see the graph with all entities
2. Go to the `/entities?entityTypeIdOrBaseUrl=https://hash.ai/@hash/types/entity-type/user/` to view all `User` entities and their linked entities, or try out a different type

## 📹 Demo




https://github.com/hashintel/hash/assets/42802102/d2f2795b-04d9-4769-92ad-44030c1b04b6
